internal/plan/plan.go

```diff
@@ -225,12 +225,12 @@ func (l *Layer) order(names []string, stop bool) ([]string, error) {
 	return order, nil
 }
 
-func (l *Layer) CheckCycles() error {
+func (l *Layer) checkCycles() error {
```

Comment:
One of the reasons this was public is that we need to make sure our effective in-memory configuration never gets corrupted to a point where running it is impossible. I believe we have introduced that possibility now that we can mutate the layer stack without this being called. We can either make this public and call it, or we might make CombineLayers run the check itself.

Not sure about which way is best. Can you please think a bit about it and implement the best approach? As long as we don't allow the issue, we can always tune later.

Reply:
Yeah, good call. I still don't think it should be exported, as callers shouldn't have to remember to do that -- the parse/combine functions should just check for it and return error. So I've kept it unexported but made `CombineLayers` check (and added a test for this case).

So now both the functions that create layers (parse and combine) check this. A caller could still bypass it by instantiating a `&plan.Layer{...}` with cycles directly, but I think that's an okay tradeoff, and kind of unavoidable with how the type is defined with all exported fields.

Reply:
Sounds good, thanks!